src/braket/tasks/local_quantum_task.py

```diff
@@ -30,38 +49,280 @@ class LocalQuantumTask(QuantumTask):
+        inputs_copy.update(kwargs["inputs"])
+        program = Program(
+            source=program.source,
+            inputs=inputs_copy,
+        )
+    return [program, shots]
+
+
+@convert_to_sim_format.register
+def _(circuit: Circuit, delegate, shots: Optional[int] = None, **kwargs) -> list:  # noqa
+    simulator = delegate
+    if DeviceActionType.OPENQASM in simulator.properties.action:
+        validate_circuit_and_shots(circuit, shots)
+        program = circuit.to_ir(ir_type=IRType.OPENQASM)
+        program.inputs.update(kwargs["inputs"] or {})
+        return [program, shots]
+    elif DeviceActionType.JAQCD in simulator.properties.action:
+        validate_circuit_and_shots(circuit, shots)
+        program = circuit.to_ir(ir_type=IRType.JAQCD)
+        qubits = circuit.qubit_count
+        return [program, qubits, shots]
+    raise NotImplementedError(f"{type(simulator)} does not support qubit gate-based programs")
+
+
+@convert_to_sim_format.register
+def _(problem: Problem, delegate, shots: Optional[int] = None, **kwargs) -> list:  # noqa
+    simulator = delegate
+    if DeviceActionType.ANNEALING not in simulator.properties.action:
+        raise NotImplementedError(f"{type(simulator)} does not support quantum annealing problems")
+    ir = problem.to_ir()
+    return [ir, shots]
+
+
+@convert_to_sim_format.register
+def _(
+    program: AnalogHamiltonianSimulation, delegate, shots: Optional[int] = None, **kwargs  # noqa
+) -> list:
+    simulator = delegate
+    if DeviceActionType.AHS not in simulator.properties.action:
+        raise NotImplementedError(
+            f"{type(simulator)} does not support analog Hamiltonian simulation programs"
+        )
+    return [program.to_ir(), shots]
+
+
+@convert_to_sim_format.register
+def _(program: AHSProgram, delegate, shots: Optional[int] = None, **kwargs) -> list:  # noqa
+    simulator = delegate
+    if DeviceActionType.AHS not in simulator.properties.action:
+        raise NotImplementedError(
+            f"{type(simulator)} does not support analog Hamiltonian simulation programs"
+        )
+    return [program, shots]
+
+
+@singledispatch
+def wrap_results(
+    results: Union[GateModelTaskResult, AnalogHamiltonianSimulationTaskResult, AnnealingTaskResult]
+) -> [
+    GateModelQuantumTaskResult,
+    AnnealingQuantumTaskResult,
+    AnalogHamiltonianSimulationQuantumTaskResult,
+]:
+    raise NotImplementedError
+
+
+@wrap_results.register
+def _(results: GateModelTaskResult) -> GateModelQuantumTaskResult:
+    return GateModelQuantumTaskResult.from_object(results)
+
+
+@wrap_results.register
+def _(
+    results: AnalogHamiltonianSimulationTaskResult,
+) -> AnalogHamiltonianSimulationQuantumTaskResult:
+    return AnalogHamiltonianSimulationQuantumTaskResult.from_object(results)
+
+
+@wrap_results.register
+def _(results: AnnealingTaskResult) -> AnnealingQuantumTaskResult:
+    return AnnealingQuantumTaskResult.from_object(results)
```

Comment:
Should I move this to a new file or a different file ?

Reply:
If you end up creating a file in braket.tasks with shared code between aws/local tasks, this would probably go there, but I don't mind moving it later. We should it make it a private function `_wrap_results` though, so we can move it/alter it without it being a breaking change

Reply:
Gotcha.